File: examples/src/Shapes/ExtractEmbeddedFileDataFromOLEObject.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using System
using System.Collections.Generic
using System.IO
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.shapes
{
    class ExtractEmbeddedFileDataFromOLEObject
    {
        public static void Run() {

            #ExStart:ExtractEmbeddedFileDataFromOLEObject

            # The documents directory path.
            dataDir = RunExamples.GetDataDir_Shapes()

            pptxFileName = dataDir +"TestOlePresentation.pptx"
            using (Presentation pres = new Presentation(pptxFileName))
            {
                objectnum = 0
                foreach (sld in pres.Slides)
                {
                    foreach (IShape shape in sld.shapes)
                    {
                        if (shape is OleObjectFrame)
                        {
                            objectnum++
                            OleObjectFrame oleFrame = shape as OleObjectFrame
                            byte[] data = oleFrame.EmbeddedData.EmbeddedFileData
                            fileExtention = oleFrame.EmbeddedData.EmbeddedFileExtension

                            extractedPath = dataDir +"ExtractedObject_out" + objectnum + fileExtention
                            using (FileStream fs = new FileStream(extractedPath, FileMode.Create))
                            {
                                fs.Write(data, 0, data.Length)
                            }
                        }
                    }
                }
            }

            #ExEnd:ExtractEmbeddedFileDataFromOLEObject

        }
    }
}

File: examples/src/shapes/extract_embedded_file_data_from_ole_object.py
import aspose.slides as slides


def extract_embedded_file_data_from_ole_object(global_opts):
    with slides.Presentation(global_opts.data_dir + "shapes_ole_objects.pptx") as pres:
        object_num = 0
        for slide in pres.slides:
            for shape in slide.shapes:
                if type(shape) is slides.OleObjectFrame:
                    object_num += 1
                    data = shape.embedded_data.embedded_file_data
                    extension = shape.embedded_data.embedded_file_extension

                    file_name = "shapes_ole_objects{idx}_out{ex}".format(idx=object_num, ex=extension)
                    with open(global_opts.out_dir + file_name, "wb") as fs:
                        fs.write(data)
